### PR TITLE
ADR for paramaterising terraform roots

### PR DESCRIPTION
**Pull request by alice-carr**
Trello Ticket: https://trello.com/c/QC5D7hgL/317-write-an-adr-for-terraform-restructuring

# PR Checklist

- [x] If you are proposing a new decision record document, used the right template for that
      - ([ADR](https://github.com/alphagov/forms/blob/main/ADR/ADRXXX-architecture-decision-record-template.md), [decision-record](https://github.com/alphagov/forms/blob/main/decision-record/DRXXX-decision-record-template.md), engagement, [research](https://github.com/alphagov/forms/blob/main/research/YYYY-MM-DD-template.md))
- [x] Set yourself as the Assignee
- [ ] Tag anyone you would like to review, or @forms-design or @forms-devs
- [ ] Fill in the template below

### CODE REVIEW

File: ADR/ADR023-paramaterise-common-terraform-roots.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+## Consequences
+
+### Pros
+* creating a new environment with this terraform structure will be quicker and easier, since there is less configuration that will need to be done
+* we have removed repeated code, making the terraform easier to read
+* this reduces the chance that environments will differ due to human error, since paramaterised inputs are stored together within the root
+
+### Cons
+* manually deploying terraform is now a bit more complicated. We have remedied this with the addition of a [makefile](https://github.com/alphagov/forms-deploy/blob/main/Makefile) which can now be used to run terraform locally
```
Comment (danworth) [inline ADR/ADR023-paramaterise-common-terraform-roots.md:88]: I agree with all the pros and this is a nicely written ADR. I mostly agree with this con but I think its currently underplaying it a bit. I wouldn't say that the additional complexity has been "remedied" as such. The make file and associated bash script tooling makes it _easier_ to perform the terraform operations by encapsulating the additional complexity, but it still remains and the tooling will need to be understood, maintained and modified as necessary going forward.

Comment (alice-carr) [reply ADR/ADR023-paramaterise-common-terraform-roots.md]: thanks Dan - I've updated this

**Review by cadmiumcat — APPROVED** [review]
I think this reflects well what we have done

File: ADR/ADR023-paramaterise-common-terraform-roots.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+# ADR023: Paramaterise Common Terraform Roots
```
Comment (aliuk2012) [inline ADR/ADR023-paramaterise-common-terraform-roots.md:1]: ADR23 has already been taken along with ADR24 which is in another PR that Laurence has created

**Review by AP-Hunt — DISMISSED** [review]
Good job capturing the cons

File: ADR/ADR023-paramaterise-common-terraform-roots.md (hunk 2 above)
Comment (alice-carr) [reply ADR/ADR023-paramaterise-common-terraform-roots.md]: thanks Al, I've updated the number

**Review by AP-Hunt — APPROVED** [review]

**Pull request merged**